src/taipy/config/global_app/global_app_config.py

```diff
@@ -111,7 +111,11 @@ def repository_properties(self, val):
 
     @property
     def properties(self):
-        return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
+        return (
+            {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
+            if self._repository_properties
+            else self._DEFAULT_REPOSITORY_PROPERTIES
```

Comment:
```suggestion
            else dict()
```
when we set `_DEFAULT_REPOSITORY_PROPERTIES` at the beginning of the class, `dict()` will return a pointer. So whenever we return `self._DEFAULT_REPOSITORY_PROPERTIES` here, we are essentially returning a pointer that was created prior to this call. Every changes made to the returned dictionary, will affect `self._DEFAULT_REPOSITORY_PROPERTIES` and this will cause bugs in the future :D

Reply:
Example to demonstrate what I mean:
```
DEFAULT = dict()

def get_dict():
        return DEFAULT

print(get_dict()) # {}

var = get_dict()
var['hi'] = 'hello'

print(var) # {'hi': 'hello'}
print(DEFAULT) # {'hi': 'hello'}
print(get_dict()) # {'hi': 'hello'}
```
In contrast with:
```
def get_dict():
        return dict()

print(get_dict()) # {}

var = get_dict()
var['hi'] = 'hello'

print(var) # {'hi': 'hello'}
print(get_dict()) # {}
```

Reply:
I noticed that the `default_config()` in the same file also has the same issue. Please fix it if you have the time :D Thanks!

Reply:
Indeed, these should be constant.

I think we need to check for these constant changes in core too. This problem might exist somewhere else.

Reply:
I believe we should return `copy(_DEFAULT_REPOSITORY_PROPERTIES)` instead of `_DEFAULT_REPOSITORY_PROPERTIES`.

We need to check the default dict usages in the config. Yes.

Reply:
In that case, we have to make sure _DEFAULT_REPOSITORY_PROPERTIES is __DEFAULT_REPOSITORY_PROPERTIES instead, so that it won't be exposed and changed by any external factors. What do you think?

Reply:
Should we use `copy.deepcopy()`?
This will not hurt the performance too much since it will not be used frequently.